hack/generate.sh

```diff
@@ -71,9 +72,15 @@ function generate_configuration() {
     sed -i "s|OPENSHIFT_VERSIONS=.*|OPENSHIFT_VERSIONS=${OPENSHIFT_VERSIONS}|" ${__root}/config/onprem-iso-fcc.yaml
     sed -i "s|PUBLIC_CONTAINER_REGISTRIES=.*|PUBLIC_CONTAINER_REGISTRIES=${PUBLIC_CONTAINER_REGISTRIES}|" ${__root}/config/onprem-iso-fcc.yaml
 
-    sed -i "s|value: '.*' # openshift version|value: '${OPENSHIFT_VERSIONS}' # openshift version|" ${__root}/config/manager/manager.yaml
     sed -i "s|HW_VALIDATOR_REQUIREMENTS=.*|HW_VALIDATOR_REQUIREMENTS=${HW_VALIDATOR_REQUIREMENTS}|" ${__root}/config/onprem-iso-fcc.yaml
     docker run --rm -v ${__root}/config/onprem-iso-fcc.yaml:/config.fcc:z quay.io/coreos/fcct:release --pretty --strict /config.fcc > ${__root}/config/onprem-iso-config.ign
+
+    # Updated operator manifests with openshift versions
+    sed -i "s|value: '.*' # openshift version|value: '${OPERATOR_OPENSHIFT_VERSIONS}' # openshift version|" ${__root}/config/manager/manager.yaml
+    # This python one-line is responsible for updating the sample AgentServiceConfig to include the latest + correct osImages
+    # When the CSV is built, this is included in the `almExamples` so that when a user goes through the OpenShift console
+    # to create the `agent` this will give them the correct defaults.
+    python3 -c "import json, sys, yaml; f = open('${__root}/config/samples/agent-install.openshift.io_v1beta1_agentserviceconfig.yaml', 'r+'); doc = yaml.load(f, Loader=yaml.FullLoader); doc['spec']['osImages'] = [{'openshiftVersion': k, 'version': v['rhcos_version'], 'url': v['rhcos_image'], 'rootFSUrl': v['rhcos_rootfs']} for (k,v) in json.loads('${OPERATOR_OPENSHIFT_VERSIONS}').items()]; f.seek(0); f.truncate(); yaml.dump(doc, f)"
```

Comment:
You can let it span multiple lines like in [here](https://github.com/openshift/release/blob/3f722a154be48991c3f65299b7c7249cb2a09b9a/ci-operator/step-registry/single-node/conf/aws/single-node-conf-aws-commands.sh#L12)